app/dao/services_dao.py

```diff
@@ -263,3 +263,31 @@ def dao_fetch_todays_stats_for_all_services(include_from_test_key=True):
         query = query.filter(Notification.key_type != KEY_TYPE_TEST)
 
     return query
+
+
+@statsd(namespace='dao')
+def fetch_stats_by_date_range_for_all_services(start_date, end_date, include_from_test_key=True):
+    query = db.session.query(
+        NotificationHistory.notification_type,
+        NotificationHistory.status,
+        NotificationHistory.service_id,
+        func.count(NotificationHistory.id).label('count')
+    ).select_from(
+        Service
+    ).join(
+        NotificationHistory
+    ).filter(
+        func.date(NotificationHistory.created_at) >= start_date,
+        func.date(NotificationHistory.created_at) <= end_date
+    ).group_by(
+        NotificationHistory.notification_type,
+        NotificationHistory.status,
+        NotificationHistory.service_id
+    ).order_by(
+        NotificationHistory.service_id
+    )
+
+    if not include_from_test_key:
+        query = query.filter(NotificationHistory.key_type != KEY_TYPE_TEST)
+
+    return query
```

Comment:
can you make this return `query.all()`?

i wouldn't mind (and indeed, would even encourage it as it results in less data being held in memory), except for the fact we're wrapping this in the `@statsd` timing decorator, and the query won't actually execute until you call .all on it, leading to the timing being misleading otherwise